### PR TITLE
[BFCL] Patch Generation Script for Locally Hosted OSS model

### PR DESCRIPTION
**Pull request by HuanzhiMao**
This PR updates the `inference` method in the `oss_handler` to improve the generation task for locally-hosted models. Previously, we used Ray for multi-node single-GPU inference (pipeline parallel), which is uncommon as most setups are single machines with multiple GPUs. This approach also led to out-of-memory errors for some large models. 

In this PR:
- Ray is removed and replaced with vllm's built-in single-node multi-GPU inference method (tensor parallel).
- The use of the `@torch.inference_mode()` decorator has been removed.

### CODE REVIEW

File: berkeley-function-call-leaderboard/model_handler/oss_handler.py [hunk 1]
```diff
@@ -34,78 +29,78 @@ def _format_prompt(prompt, function, test_category):
...
+        test_question,
         model_path,
         temperature,
         max_tokens,
         top_p,
-        format_prompt_func,
-        index,
+        stop_token_ids=None,
+        max_model_len=None,
+        num_gpus=8,
```
Comment (Fanjia-Yan) [inline berkeley-function-call-leaderboard/model_handler/oss_handler.py:41]: nit: `num_gpus` in `oss_handler` and `openfunctions_evaluation` are inconsistent.

**Review by Fanjia-Yan — APPROVED** [review]

File: berkeley-function-call-leaderboard/model_handler/oss_handler.py (hunk 1 above)
Comment (HuanzhiMao) [reply berkeley-function-call-leaderboard/model_handler/oss_handler.py:41]: ah yes, good catch

Comment (HuanzhiMao) [reply berkeley-function-call-leaderboard/model_handler/oss_handler.py]: fixed

**Comment by HuanzhiMao** [comment]
Sorry, this PR was accidentally closed.

**Comment by HuanzhiMao** [comment]
Sorry, this PR was accidentally closed.

**Comment by HuanzhiMao** [comment]
Sorry, this PR was accidentally closed.

**Review by CharlieJCJ — APPROVED** [review]
LGTM

**Review by ShishirPatil — APPROVED** [review]

**Pull request merged**